bin/build-locales

```diff
@@ -64,66 +64,94 @@ try {
-var files = glob.sync(path.resolve(__dirname, '../src/views/**/l10n.json'));
-files.forEach(function (file) {
-    var dirPath = file.split('/');
-    dirPath.pop();
-    var view = dirPath.pop();
-    localeCompare.getIdsForView(view, file, viewLocales, idsWithICU, icuWithIds);
-});
+        try {
+            fs.accessSync(path.resolve(__dirname, '../src/views/' + routes[v].view + '.jsx'));
+        } catch (err) {
+            // the config for the view is not set up correctly, so throw the error.
+            throw err;
+        }
+    }
 
-// get asset url translations
-var localizedAssetUrls = {};
-files = glob.sync(path.resolve(__dirname, '../src/views/**/l10n-static.json'));
-files.forEach(function (file) {
-    var dirPath = file.split('/');
-    dirPath.pop();
-    var view = dirPath.pop();
-    localizedAssetUrls[view] = {};
-
-    var assetUrls = JSON.parse(fs.readFileSync(file, 'utf8'));
-    for (var lang in localizedUrls) {
-        localizedAssetUrls[view][lang] = {};
-        for (var key in assetUrls) {
-            if (localizedUrls[lang].hasOwnProperty(key)) {
-                localizedAssetUrls[view][lang][key] = localizedUrls[lang][key];
-            } else {
-                localizedAssetUrls[view][lang][key] = assetUrls[key];
+    // get asset url translations
+    try {
+        subdir = routes[v].view.split('/');
+        subdir.pop();
+        var l10nStatic = path.resolve(__dirname, '../src/views/' + subdir.join('/') + '/l10n-static.json');
+        localizedAssetUrls[routes[v].name] = {};
+
+        var assetUrls = require(l10nStatic);
+        for (var lang in localizedUrls) {
+            localizedAssetUrls[routes[v].name][lang] = {};
+            for (var key in assetUrls) {
+                if (localizedUrls[lang].hasOwnProperty(key)) {
+                    localizedAssetUrls[routes[v].name][lang][key] = localizedUrls[lang][key];
+                } else {
+                    localizedAssetUrls[routes[v].name][lang][key] = assetUrls[key];
+                }
             }
         }
+    } catch (err) {
+        if (err.code !== 'MODULE_NOT_FOUND') {
+            throw err;
+        }
+
+        try {
+            fs.accessSync(path.resolve(__dirname, '../src/views/' + routes[v].view + '.jsx'));
+        } catch (err) {
+            // the config for the view is not set up correctly, so throw the error.
+            throw err;
+        }
     }
-});
+}
 
 // md5 of english strings with message key as the value for searching po files.
 // Sample structure: { 'sdfas43534sdfasdf': 'general-general.blah', 'lkjfasdf4t342asdfa': 'about-about.blah' }
 var md5WithIds = localeCompare.getMD5Map(icuWithIds);
 
 // Get ui localization strings first
 var isoCodes = Object.keys(languages);
-for (i in isoCodes) {
+for (var i in isoCodes) {
     var translations = localeCompare.getTranslationsForLanguage(isoCodes[i], idsWithICU, md5WithIds);
-    for (var key in translations) {
+    for (key in translations) {
```

Comment:
Is this to avoid a `no-redeclare` linter error? In that case I would keep the `var` and just call `key` something distinct and more descriptive. Same with `i` above, if these were called `isoCode` and `messageId`, I think it would help some of the complexity of this script if these names were more descriptive. This code is a little hard to follow if you don't know what it does already since it's transforming and consolidating a bunch of different types of objects. If the loops that operated on these described the data they were working with, I think it would help make it more understandable, and would keep you from needing to scrolling around to check the comments for what the structure of the objects are.